File: Solutions/1598/1598.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/9/9 10:33
# File: 1598.py
# Desc: 

class Solution:
    def minOperations(self, logs: List[str]) -> int:
        idx = 0
        while idx < len(logs):
            temp = logs[idx]
            if temp == './': logs.pop(idx)
            elif temp == "../":
                if idx > 0:
                    logs.pop(idx-1)
                    logs.pop(idx-1)
                    idx -= 1
                else:
                    logs.pop(idx)
            else:
                idx += 1
        return len(logs)

File: Solutions/1598/1598.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1598.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-9-9 11:13:36
***************************************************************************************************/

int minOperations(char ** logs, int logsSize){
    int ret = 0;
    for (int i=0; i<logsSize; i++) {
        if (strcmp(logs[i], "../") == 0) {
            if (ret > 0) ret--;
        } else if (strcmp(logs[i], "./") != 0) ret++;
    }
    return ret;
}
